client/app/bundles/course/assessment/submission/containers/ProgrammingImportEditor.jsx

```diff
@@ -254,7 +254,8 @@ VisibleProgrammingImportEditor.propTypes = {
 
 function mapStateToProps(state, ownProps) {
   const { questionId, answerId, question, readOnly, viewHistory } = ownProps;
-  const { submission, dispatch, submissionFlags, history } = state;
+  const { submission, dispatch, submissionFlags, history } =
```

Comment:
Should dispatch be retrieved from state or the nested store?

Reply:
This is wrong. There's no `dispatch` in `state` to begin with. This was probably never caught because `connect` provides the correct `dispatch` to the component.